### PR TITLE
feat(outlet): switch to using structural directive

### PR DESCRIPTION
**Pull request by waterplea**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/)
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Other... Please describe:

## What is the current behavior?
Currently outlet is an attribute component meaning it requires a DOM element to be used.

## What is the new behavior?
With cleanup done to ViewRef in Angular 12 it is now easy enough to make our own structural directive instead of delegating it to built-in Angular directives. This means we no longer need a component and a DOM element.

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

## Other information
We need to think if there's a way to enable primitive value by default, rather than writing following all the time:
```html
<ng-container *polymorpheusOutlet="content as text">{{text}}</ng-container>
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,16 +44,16 @@
...
-        "@angular/common": "9.1.13",
-        "@angular/compiler": "9.1.13",
-        "@angular/core": "9.1.13",
-        "@angular/forms": "9.1.13",
-        "@angular/platform-browser": "9.1.13",
-        "@angular/platform-browser-dynamic": "9.1.13",
-        "@angular/platform-server": "9.1.13",
-        "@angular/router": "9.1.13",
-        "@nguniversal/express-engine": "^9.1.1",
+        "@angular/animations": "12.0.0",
```
Comment (MarsiBarsi) [inline package.json:47]: why don't we use "^12.0.0"?

File: projects/ng-polymorpheus/tsconfig.lib.json [hunk 2]
```diff
@@ -15,7 +15,8 @@
         "fullTemplateTypeCheck": true,
         "strictInjectionParameters": true,
         "enableResourceInlining": true,
-        "enableIvy": false
+        "enableIvy": true,
+        "compilationMode": "partial"
```
Comment (MarsiBarsi) [inline projects/ng-polymorpheus/tsconfig.lib.json:19]: What is `partial`? Is there `full-100%-ivy mode`?

File: tsconfig.json [hunk 3]
```diff
@@ -5,9 +5,8 @@
         "outDir": "./dist/out-tsc",
         "sourceMap": true,
         "declaration": false,
-        "module": "esnext",
+        "module": "es2020",
```
Comment (MarsiBarsi) [inline tsconfig.json:8]: the future is now...

File: projects/ng-polymorpheus/tsconfig.lib.json (hunk 2 above)
Comment (waterplea) [reply projects/ng-polymorpheus/tsconfig.lib.json:19]: No. That is how you are supposed to compile Ivy libraries that go to npm:
https://blog.lacolaco.net/2021/02/angular-ivy-library-compilation-design-in-depth-en/

File: package.json (hunk 1 above)
Comment (waterplea) [reply package.json:47]: That's how ng update did it. I think it doesn't matter much :)

File: tsconfig.json (hunk 3 above)
Comment (waterplea) [reply tsconfig.json:8]: image

**Pull request merged**